### PR TITLE
Fix problem with spaces in paths in archivist

### PR DESCRIPTION
**Pull request by alicefuzier**
For #3193

### CODE REVIEW

File: archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala [hunk 1]
```diff
@@ -23,6 +23,14 @@ trait RandomThings {
     Random.alphanumeric take length mkString
   }
 
+  def randomAlphanumericWithSpaces(length: Int = 8) = {
```
Comment (mklander) [inline archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala]: did I read this right, is it `randomAlphanumericWithSpace`?

File: archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala [hunk 2]
```diff
@@ -23,6 +23,14 @@ trait RandomThings {
...
+    val str = randomAlphanumeric(length).toCharArray
+
+    val spaceIndex = Random.nextInt(str.length - 2) + 1
```
Comment (mklander) [inline archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala]: I guess the `-2) +1` is to not add spaces at the end or beginning?

File: archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala [hunk 3]
```diff
@@ -23,6 +23,14 @@ trait RandomThings {
...
+    str.update(spaceIndex, ' ')
+    str.toString
```
Comment (mklander) [inline archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala]: How about `str.updated(spaceIndex, ' ')`?

File: archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala (hunk 1 above)
Comment (alicefuzier) [reply archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala]: Yes, it is, will change

File: archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala (hunk 2 above)
Comment (alicefuzier) [reply archive/common/src/test/scala/uk/ac/wellcome/platform/archive/common/fixtures/RandomThings.scala]: Yes, I'll add a comment explaining

**Pull request merged**